### PR TITLE
HDDS-8802. Added pagination support for ListSnapshotDiff jobs

### PR DESCRIPTION
**Pull request by hemantk-12**
## What changes were proposed in this pull request?
This change is to provide the pagination logic to List SnapshotDiff jobs API.

## What is the link to the Apache JIRA
HDDS-8802

## How was this patch tested?
Existing unit test and manually tested on docker on local machine.

### CODE REVIEW

**Review by swamirishi — COMMENTED** [review]
Overall the patch looks good to me. Currently the patch has merge conflicts and that needs to be resolved.

**Comment by peterxcli** [comment]
Hi @hemantk-12 just curious why this PR is closed?

**Comment by hemantk-12** [comment]
I think it was closed by mistake because I deleted branch HDDS-8802 on my fork. Let me restore it and rebase it.

**Comment by hemantk-12** [comment]
Cloudn't reopen it. Created new PR #8124.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java [hunk 1]
```diff
@@ -423,30 +424,46 @@ public CancelSnapshotDiffResponse cancelSnapshotDiff(
...
           }
         }
+        if (jobs.size() == maxEntries) {
+          break;
+        }
+      }
+
+      // If maxEntries are populated and list still has more entries,
+      // set the continuation token for the next page request otherwise null.
+      if (jobs.size() == maxEntries && iterator.hasNext()) {
```
[the remark this answers is not in the document]
Comment (hemantk-12) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java:461]: Removed `job.size() == maxEntries` check in #8124

**Pull request closed without merging**